apps/testing/src/main/java/com/akto/test_editor/filter/FilterAction.java

```diff
@@ -285,6 +285,9 @@ public DataOperandsFilterResponse applyFilterOnPayload(FilterActionRequest filte
 
         BasicDBObject payloadObj = new BasicDBObject();
         try {
+            if (payload != null && payload.startsWith("[")) {
```

Comment:
i think we would need this in extract method as well for req/resp payloads, can you check once